instances/widgets.treasury-factory.near/widget/pages/asset-exchange/OneClickExchangeForm.jsx

```diff
@@ -449,151 +578,97 @@ const getAvailableNetworks = () => {
-  const outputToken = allTokensOut.find(
+  // Find the selected token info to get the symbol
+  const selectedTokenIn = intentsTokensIn.find((t) => t.id === tokenIn);
+  // tokenOut is the symbol, and we need to match it with the selected network
+  const selectedTokenOut = allTokensOut.find(
     (t) => t.symbol === tokenOut && t.network === networkOut
   );
-  if (!outputToken) return null;
 
-  return { inputToken, outputToken };
-};
+  // Don't submit if we can't find the token info
+  if (!selectedTokenIn || !selectedTokenOut) {
+    console.error("Cannot find token information for:", tokenIn, tokenOut);
+    setErrorApi("Cannot find token information. Please try again.");
+    return;
+  }
 
-// Fetch quote from 1Click API
-const fetchQuote = () => {
-  const tokenDetails = getTokenDetails();
-  if (!tokenDetails) return;
+  // Get the network name from the available networks
+  const selectedNetwork = getAvailableNetworks().find(
+    (n) => n.id === networkOut
+  );
+  const networkName = selectedNetwork ? selectedNetwork.name : networkOut;
 
-  const { inputToken, outputToken } = tokenDetails;
+  // Set loading state
+  setIsLoading(true);
+  setErrorApi(null);
 
-  // Convert amount to smallest unit
+  // Convert amount to smallest unit if needed
+  const decimals = selectedTokenIn.decimals || 18;
   const amountInSmallestUnit = Big(amountIn)
-    .mul(Big(10).pow(inputToken.decimals))
+    .mul(Big(10).pow(decimals))
     .toFixed(0);
 
-  setIsLoadingQuote(true);
-  setQuote(null);
-  setErrorApi(null);
-  setShowQuoteDetails(false);
-
-  // Calculate deadline (7 days for DAO voting)
-  const deadline = new Date();
-  deadline.setDate(deadline.getDate() + 7);
-
-  // Prepare the quote request in the format expected by 1Click API v0
-  const quoteRequest = {
-    dry: false,
-    swapType: "EXACT_INPUT",
-    slippageTolerance: parseInt(slippageTolerance), // Convert to number
-    originAsset: inputToken.id.startsWith("nep141:")
-      ? inputToken.id
-      : `nep141:${inputToken.id}`,
-    depositType: "INTENTS",
-    destinationAsset: outputToken.id,
-    refundTo: treasuryDaoID,
-    refundType: "INTENTS",
-    recipient: treasuryDaoID, // Swapped tokens stay in treasury's NEAR Intents
-    recipientType: "INTENTS",
-    deadline: deadline.toISOString(),
-    amount: amountInSmallestUnit,
+  // Prepare the request for our custom backend endpoint
+  const backendRequest = {
+    treasuryDaoID,
+    inputToken: selectedTokenIn,
+    outputToken: selectedTokenOut,
+    amountIn: amountInSmallestUnit,
+    slippageTolerance: parseInt(slippageTolerance),
+    networkOut: networkName,
   };
 
-  console.log("Fetching quote:", quoteRequest);
+  console.log("Fetching actual quote from backend:", backendRequest);
 
-  asyncFetch("https://1click.chaindefuser.com/v0/quote", {
+  // Use our custom backend endpoint to get the actual quote with proposal payload
+  asyncFetch("${REPL_BACKEND_API}/treasury/oneclick-quote", {
```

Comment:
so what's the dry quote and this?
wouldn't there be any difference between them, since they are called at different time, the rate can change and also the difference in fees, just want to make sure we are showing the correct rate to the user

Reply:
According to the docs we should use `dry: true` to display the quote price without generating a transaction, and only `dry: false` at confirmation to reduce system load ( https://docs.near-intents.org/near-intents/integration/distribution-channels/1click-api#post-v0-quote ).

I agree though that we should not just generate the real quote without also confirming that. The real quote will generate a deposit address, and the proposal to the DAO will simply be about depositing to that address. We should also show this information clearly before creating the proposal. Do you have some UX recommendations for this @olskik12 ?

Reply:
Opened the issue here: https://github.com/NEAR-DevHub/neardevhub-treasury-dashboard/issues/681